agent/lib/kontena/websocket_client.rb

```diff
@@ -166,12 +179,15 @@ def on_close(event)
       @connected = false
       @connecting = false
       @ws = nil
-      if event.code == 4001
+
+      case event.code
+      when 4001
         handle_invalid_token
-      elsif event.code == 4010
-        handle_invalid_version
+      when 4010
+        handle_invalid_version(event.reason)
+      else
```

Comment:
I think it makes sense to abort on 404x conditions. I.e. the docker ID conflict will not resolve by itself and a hint in logs would help a lot.